### ISSUE TITLE
Cannot upgrade from 4.3.0RC3 to 4.3.0RC4 (cakephp/queue container dep)

### ISSUE THREAD

**Issue by CriztianiX** [cakephp/cakephp#16046]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 4.3.0RC4
* Platform and Target: Docker, PHP8.1RC4

### What you did
I've tried to upgrade from 4.3.0rc3 to 4.3.0rc4
`php composer.phar require  "cakephp/cakephp: 4.3.0-RC4"`

### What happened
cakephp/queue is using psr/continer v1, in cakephp 4.3.0rc4 you upgraded to psr/container v2. The upgrade cannot continue due to locked version.

```
  Problem 1
    - enqueue/enqueue 0.10.9 requires psr/container ^1 -> found psr/container[1.0.0, 1.1.0, 1.1.1] but it conflicts with your root composer.json require ( 2.0.1).
    - cakephp/queue 0.1.3 requires enqueue/simple-client ^0.10 -> satisfiable by enqueue/simple-client[0.10.9].
    - enqueue/simple-client 0.10.9 requires enqueue/enqueue ^0.10 -> satisfiable by enqueue/enqueue[0.10.9].
    - cakephp/queue is locked to version 0.1.3 and an update of this package was not requested.
```

### What you expected to happen
Upgrade to 4.3.0RC4 without issues.

**Comment by othercorey** [comment]
@ADmad I don't see a compatible version of enqueue.

**Comment by CriztianiX** [comment]
> @ADmad I don't see a compatible version of enqueue.

Exactly. 
IMO 4.3.0 should be using to psr/container v1 again.
Maybe container v2 should be in 4.5.0

**Comment by othercorey** [comment]
If we keep v1 for 4.3 that will probably be the version used until cakephp 5.

**Comment by CriztianiX** [comment]
are there some breaking changes between v1 and v2? maybe we can make a PR to https://github.com/php-enqueue/enqueue-dev
with something like:

```
    "psr/log": "^1.1 || ^2.0 || ^3.0",
    **"psr/container": "^1 || ^2.0",**
```

**Comment by ADmad** [comment]
> are there some breaking changes between v1 and v2?

The only change is addition of return type https://github.com/php-fig/container/releases/tag/2.0.0.

So yes a PR to ` php-enqueue/enqueue-dev` updating the constraint to `^1  || ^2` for `psr/container` should suffice.

**Comment by ADmad** [comment]
Given that the latest release of `enqueue/enqueue` is PHP 7.3+, it should absolutely allow using `psr/container` v2.

**Comment by CriztianiX** [comment]
PR at https://github.com/php-enqueue/enqueue-dev/pull/1205

**Comment by ADmad** [comment]
Mine is more complete :slightly_smiling_face:  https://github.com/php-enqueue/enqueue-dev/pull/1206

**Comment by ADmad** [comment]
Closing as there's nothing to be done here except waiting for the php-enque devs to merge the related PR.

**Comment by othercorey** [comment]
php-enqueue released 0.10.15 with this change.
